monkey/monkey_island/cc/models/monkey.py

```diff
@@ -0,0 +1,57 @@
+"""
+Define a Document Schema for the Monkey document.
+"""
+import mongoengine
+from mongoengine import Document, StringField, ListField, BooleanField, EmbeddedDocumentField, DateField, \
+    ReferenceField
+
+from monkey_island.cc.models.errors import MonkeyNotFoundError
+from monkey_island.cc.models.monkey_ttl import MonkeyTtl
+
+
+class Monkey(Document):
+    """
+    This class has 2 main section:
+        *   The schema section defines the DB fields in the document. This is the data of the object.
+        *   The logic section defines complex questions we can ask about a single document which are asked multiple
+            times, somewhat like an API.
+    """
+    # SCHEMA
+    guid = StringField(required=True)
+    config = EmbeddedDocumentField('Config')
+    creds = ListField(EmbeddedDocumentField('Creds'))
+    dead = BooleanField()
+    description = StringField()
+    hostname = StringField()
+    internet_access = BooleanField()
+    ip_addresses = ListField(StringField())
+    keepalive = DateField()
+    modifytime = DateField()
+    # TODO change this to an embedded document as well - RN it's an unnamed tuple which is confusing.
```

Comment:
This will require further changes to the way out DB is structured, and isn't an integral part of the monkey timeout feature. Why does this need to block monkey timeout?

Reply:
Can you elaborate the changes? Just so we figure out how big this is

Reply:
There are about 12 places in the code that refer to parent, probably a small-medium change if we want to cover all of them correctly. Seems like a good thing to do just unrelated to this feature.  Most of it is in the post request in resources/monkey.py.